src/ampycloud/utils/utils.py

```diff
@@ -150,6 +159,23 @@ def check_data_consistency(pdf: pd.DataFrame,
             logger.warning('Dropping the superfluous %s column from the input data.', key)
             data.drop((key), axis=1, inplace=True)
 
+    # Check for any duplicated entry, which would make no sense.
+    if (duplic := data.duplicated()).any():
+        raise AmpycloudError('Duplicated hits in the input data:\n'
+                             f'{data[duplic].to_string(index=False)}')
+
+    # Check for inconsistencies
+    # 1 - A non-detection should not be coincident with a detection
+    # 2 - A VV hit should not be coincident with a hit or a non-detection
+    for hit_type in [0, -1]:
+        nodets = data[data['type'] == hit_type][['dt', 'ceilo']]
+        dets = data[data['type'] != hit_type][['dt', 'ceilo']]
+        merged = dets.merge(nodets, how='inner', on=['dt', 'ceilo'])
+        if len(merged) > 0:
+            raise AmpycloudError('Inconsistent input data '
+                                 f'(simultaneous type {hit_type} and !{hit_type}):\n'
+                                 f'{merged.to_string(index=False)}')
```

Comment:
```suggestion
   for hit_type in [0, -1]:
        data_hit_type = data[data['type'] == hit_type][['dt', 'ceilo']]
        data_other = data[data['type'] != hit_type][['dt', 'ceilo']]
        common_timestamps = set(data_hit_type['dt']).intersection(set(data_other['dt']))
        if len(common_timestamps) > 0:
            raise AmpycloudError(
                f'Inconsistent input data (simultaneous type {hit_type} and !{hit_type}) '
                f'for the following timestamps:\n {common_timestamps}'
            )
```

Reply:
Your change is not equivalent to the original suggestion, because you no longer account for the ceilometer name in the "intersection". I.e. it is perfectly fine to have a VV hit with ceilo A and no hit with ceilo B.

I have added an additional test for this case, such that your proposed change will no longer pass the tests ;)